packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts

```diff
@@ -106,5 +106,6 @@ export const useRunMetricExplorerQuery = ({
             !!metricName &&
             !!comparison &&
             !!dateRange,
+        keepPreviousData: true,
```

Comment:
smoother experience to the user

Reply:
we just have to be careful with isLoading usage and utilise isFetching when we do better loading states